raster/src/main/scala/geotrellis/raster/summary/polygonal/Implicits.scala

```diff
@@ -17,16 +17,30 @@
 package geotrellis.raster.summary.polygonal
 
 import geotrellis.raster._
-
-
-object Implicits extends Implicits
+import geotrellis.util.GetComponent
 
 trait Implicits {
-  implicit class withSinglebandTilePolygonalSummaryMethods(
-    val self: Tile
-  ) extends SinglebandTilePolygonalSummaryMethods
+  implicit def rasterHasRasterExtent[T <: CellGrid[Int]]
+    : GetComponent[Raster[T], RasterExtent] =
+    new GetComponent[Raster[T], RasterExtent] {
+      override def get: Raster[T] => RasterExtent = _.rasterExtent
+    }
 
-  implicit class withMultibandTilePolygonalSummaryMethods(
-    val self: MultibandTile
-  ) extends MultibandTilePolygonalSummaryMethods
+  implicit class withPolygonalSummaryResultMethods[A](val self: PolygonalSummaryResult[A]) {
+    def toEither: Either[NoIntersection.type, A] = {
+      self match {
+        case NoIntersection => Left(NoIntersection)
+        case Summary(value) => Right(value)
+      }
+    }
+
+    def toOption: Option[A] = {
```

Comment:
Both `toEither` and `toOption` should be directly on `PolygonalSummaryResult`. They make perfect sense as part of the class, given a chance to convert to standard type if more detail is not helpful.

From experience it is tempting to put things that don't seem core in implicit extension methods, but you still end up with tight binding between the method and the interface due to SemVer constraints and an obscure location for the functionality to boot.

Reply:
Roger.